middleware/compress.go

```diff
@@ -78,8 +79,9 @@ func GzipWithConfig(config GzipConfig) echo.MiddlewareFunc {
 				}
 				rw := res.Writer
 				w.Reset(rw)
+				grw := &gzipResponseWriter{Writer: w, ResponseWriter: rw}
 				defer func() {
-					if res.Size == 0 {
+					if res.Size == 0 && !grw.wroteHeader {
```

Comment:
1. `wroteHeader` name should be `wroteBody` of something like that - it is not about writing header at the moment.

2. I am not sure if this is correct solution. The whole if `if res.Size == 0 {` statement point is to remove `echo.HeaderContentEncoding` header so  empty messages would not appear as gzip responses.  But this `!grw.wroteHeader` will negate that intention.

Maybe `if !grw.wroteBody {` would make more sense. And remove unnecessary `http.StatusNoContent` check from `WriteHeader` as NoContent is when we did not write anything to client.

so it could be:
```go
func (w *gzipResponseWriter) WriteHeader(code int) {
	w.Header().Del(echo.HeaderContentLength) // Issue #444
	w.ResponseWriter.WriteHeader(code)
}
```

Reply:
I'm sorry. It is true that `wroteHeader` is not clear whether it is an http header or a gzip header, and the name was confusing.
I changed the name to `wroteBody` because the gzip header is part of the http body.

Reply:
Maybe I fixed all.